### ISSUE TITLE
[HOLD for payment 2023-05-26] [$4000] Web - Triple clicking an edited message select’s the (edited) label along with the actual message

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15194]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Launch the app -> Login in account
2. Navigate to any conversation
3. Send an message -> Edit that message (make sure that has (edited) label at end of statement)
4. Try to select that message by triple clicking on message -> Notice message is selected along with (edited) label

## Expected Result:
(edited) label should not get selected

## Actual Result:
 (edited) label get selected along with message

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.72-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/219250345-846d6e72-489f-47da-b58a-4579202c565e.mp4

https://user-images.githubusercontent.com/43996225/219250368-b1870af8-11c6-4976-9dab-e4729ea4f641.mov

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhairyasenjaliya 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1676445116253349

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d2bbf84b58c19ab3</li>
        <li>Upwork Job ID: 1628536904239013888</li>
        <li>Last Price Increase: 2023-03-29</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @kevinksullivan (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01d2bbf84b58c19ab3

**Comment by MelvinBot** [comment]
Current assignee @kevinksullivan is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
📣 @MelvinBot! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @parasharrajat (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @marcochavezf (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by kevinksullivan** [comment]
Sent to the pool.

**Comment by allroundexperts** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The edited text gets selected when message is double clicked along with the actual message.

### What is the root cause of that problem?
The `edited` text is `user-selectable` by default.

### What changes do you think we should make in order to solve the problem?

We can add the following [here](https://github.com/allroundexperts/Expensify/blob/2ce85d879f64b10096742aa4db755aa9bd12c4ab/src/components/HTMLEngineProvider/HTMLRenderers/EditedRenderer.js#L22):
`style={{userSelect: 'none'}}`

### What alternative solutions did you explore? (Optional)
None

**Comment by kavimuru** [comment]
@parasharrajat I have updated the platforms affected.

**Comment by kevinksullivan** [comment]
https://expensify.slack.com/archives/C049HHMV9SM/p1677201869393969?thread_ts=1676445116.253349&cid=C049HHMV9SM

**Comment by allroundexperts** [comment]
> Meanwhile,
> 
> To Contributors: 
> 
> 
> 
>  Is it possible to prevent quick selections to the edited label without making it unselectable? 
> 
>  

What do you mean by quick selection @parasharrajat?

**Comment by allroundexperts** [comment]
> I think triple click can be considered a quick selection where the user does not have to pick starting and ending points. 
> 
> 
> 
> Other examples might be where you double click and the word or line under the cursor will get selected automatically.
> 
> 
> 
> Does this help?

Thank you for the reply. I think this should be possible with TextSelection API. We can choose a range offset such that it does not include the last 6 characters if they form the word "edited". I will post a small fiddle of this as well shortly.

**Comment by jayeshmangwani** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue. 
When we try to copy edited messages using triple click then edited label is also being selected along with message

### What is the root cause of that problem? 
We are wrapping the message and edited label under a single text component so the whole text is being selected here , as we are passing the `selectable={!DeviceCapabilities.canUseTouchScreen() || !props.isSmallScreenWidth}` to parent Text Component, right now structure Is like this  `<Text selectable>{message}<Text>(edited)</Text></Text> `

### What changes do you think we should make in order to solve the problem?
We can extract the edited label out of parent Text component and we can use like this 
```
<View style={{flexDirection: ‘row’}>
    <Text>{message}</Text>
    <Text>(edited)</Text>
</View> 
```

What alternative solutions did you explore? (Optional)
None

**Comment by MelvinBot** [comment]
📣 @jayeshmangwani! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by allroundexperts** [comment]
> > I think triple click can be considered a quick selection where the user does not have to pick starting and ending points.
> > Other examples might be where you double click and the word or line under the cursor will get selected automatically.
> > Does this help?
> 
> Thank you for the reply. I think this should be possible with TextSelection API. We can choose a range offset such that it does not include the last 6 characters if they form the word "edited". I will post a small fiddle of this as well shortly.

@parasharrajat Here is the [sandbox](https://stackblitz.com/edit/typescript-6xxmkg?file=index.ts,index.html) that I've created. Below is a video as well of the behaviour.

https://user-images.githubusercontent.com/30054992/221322843-b92bd9fa-855d-453f-94c4-bf614574e0cb.mov

**Comment by allroundexperts** [comment]
Throwing out an idea for preventing the copying of the text. We can traverse the elements of the `Selection` [here](https://github.com/allroundexperts/Expensify/blob/2e46ecbd2e5af56cd4155cc59abbdaf8ae10a418/src/libs/SelectionScraper/index.js#L42) and filter out all elements that have `user-select` set as `none`. Probably not the most efficient way but I think this will work file.

**Comment by MelvinBot** [comment]
@marcochavezf @kevinksullivan @parasharrajat this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by s77rt** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Triple-click selection includes `(edited)` label for edited messages.

### What is the root cause of that problem?
Not a problem, this is the expected behaviour since the "edited" label is a selectable `<Text />`. This is more like a feature request.

### What changes do you think we should make in order to solve the problem?
Wrap the label ([here](https://github.com/Expensify/App/blob/a7c196fb09e06ed31dfe2f520da134515c835386/src/pages/home/report/ReportActionItemFragment.js#L128) and [here](https://github.com/Expensify/App/blob/a7c196fb09e06ed31dfe2f520da134515c835386/src/components/HTMLEngineProvider/HTMLRenderers/EditedRenderer.js#L18)) with a `<Text />` having styles `position: 'absolute'` and `width: 'max-content'`.
Absolute positioned elements are considered as separated elements for the browser.

### What alternative solutions did you explore? (Optional)
- Instead of using `position: 'absolute'` use `float: 'right'`.

___

Unfortunately, none of the above solutions are perfect, the first one being absolute positioned will not respect overflow and margins of the parent, and may be cut in the middle instead of wrapping to the next line. While the second will make the label stick to the right. Just submitting as an inspiration for better solutions :grin:.

**Comment by MelvinBot** [comment]
@marcochavezf, @kevinksullivan, @parasharrajat Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by MelvinBot** [comment]
@marcochavezf, @kevinksullivan, @parasharrajat Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by kevinksullivan** [comment]
@parasharrajat thoughts on the proposal above?

**Comment by s77rt** [comment]
@getusha Nice, I was trying to achieve something similar but sadly this does not work, it breaks triple click selection and I think @Ollyws's proposal too.


https://user-images.githubusercontent.com/16493223/223822289-a40bf960-a2bb-438c-b409-d82d8e93bbbe.mp4

___

For future proposal please make sure triple selection works on both cases:
1. Triple selection on text
2. Triple selection next to text

Expected behaviour should be something like that:

https://user-images.githubusercontent.com/16493223/223822651-38f1bd88-0ccd-47dc-bebf-7a3ba9c8781d.mp4

**Comment by s77rt** [comment]
Personally I don't see much value in having the edited label being selectable, but it would be great if we can fix this without modifying anything else. If we don't get an acceptable solution for this we may end-up either applying one of earliest proposals (making the label unselectable) or closing the issue given it's low priority/value.

**Comment by s77rt** [comment]
Heads up. We have a similar bug [here](https://github.com/Expensify/App/issues/15810).
If it's okay to disable selection on the edited label then we should do so and in addition apply the proposed fix [here](https://github.com/Expensify/App/issues/15810#issuecomment-1465023649).

Thanks @allroundexperts for pointing this out!

**Comment by tienifr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
(edited) label get selected along with message when triple clicking the message.

### What is the root cause of that problem?
The root cause is the `(edited)` label is currently selectable and copiable, so when we select/copy the texts nearby it, it will select/copy the `(edited)` label as well.

### What changes do you think we should make in order to solve the problem?


You've probably seen the Github's code line number texts, they are never selectable or copiable.
<img width="67" alt="Screenshot 2023-03-12 at 21 37 37" src="https://user-images.githubusercontent.com/113963320/224551667-68a32ba5-de03-4855-87c6-5fe936a4cef9.png">

We should implement the same approach as Github did, which is to make use of the `::before` pseudo element to set the text content, which is never selectable or copiable. The text content set by `::before` will still have all the style of the `Text` element, so we don't need any style change.

Here are the steps:
1. Introduce a new `copiable` prop to our `Text` component, defaulting to `true`
2. If `copiable` is false, we'll set the content of the text using the `::before` pseudo element, this can be done by adding the following logic  
```
if (!copiable && typeof children === 'string') {
        return (
            <RNText
                allowFontScaling={false}
                ref={ref}
                style={[componentStyle]}
                dataSet={{
                    textContent: children,
                    textCopiable: false,
                }}
                {...props}
            />
        );
    }
```
right before https://github.com/Expensify/App/blob/3b54638d85bfe9e82b4fa02d749eaf64325050d8/src/components/Text.js#L67
(if needs to, we can explore extending the logic so that it works for Text element which `children` is not a string, but for this issue the above is enough)

3. Add the CSS for the `::before` pseudo element, by adding
```
[data-text-copiable="false"]::before {
    content: attr(data-text-content)
}
```
in `<style>` in here https://github.com/Expensify/App/blob/main/web/index.html.

4. For any text that we want to be non-selectable and non-copiable, we can just add the `copiable={false}` to the `<Text>` element. For this issue it will be here https://github.com/Expensify/App/blob/5dfd58c0d853926b7d30d3eb50855e9a85d6a5b6/src/pages/home/report/ReportActionItemFragment.js#L124
and here https://github.com/Expensify/App/blob/731fbd2e08bb4f0923c9343835b92475a912631d/src/components/HTMLEngineProvider/HTMLRenderers/EditedRenderer.js#L18 (For this one, make sure to add in in the children `Text` element rather than the parent one)

### What alternative solutions did you explore? (Optional)
We can do some changes to the HTML structure so that the triple click will not copy the `edited`, but it's not a general approach, only a case-by-case fix, and it can easily break if we have updates to the HTML in the future.

### Working well after the fix
https://user-images.githubusercontent.com/113963320/224553164-f935717f-b080-470f-ba17-2fef7222083c.mov

**Comment by MelvinBot** [comment]
@marcochavezf, @kevinksullivan, @parasharrajat Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by bernhardoj** [comment]
Hi, just want to point out that the text with `user-select: none` still copyable is a Chromium [bug](https://bugs.chromium.org/p/chromium/issues/detail?id=850685) and looks like it's fixed on the latest version. I don't know about Safari though.

**Comment by tienifr** [comment]
It's very consistent across all browsers so I wouldn't say it's a bug. I can reproduce the same behavior on latest Chrome.

**Comment by allroundexperts** [comment]
> Throwing out an idea for preventing the copying of the text. We can traverse the elements of the `Selection` [here](https://github.com/allroundexperts/Expensify/blob/2e46ecbd2e5af56cd4155cc59abbdaf8ae10a418/src/libs/SelectionScraper/index.js#L42) and filter out all elements that have `user-select` set as `none`. Probably not the most efficient way but I think this will work fine.

@parasharrajat We're already making use of the `Selection` API heavily in the `SelectionScrapper` file. In there, we're duplicating ranges, changing there end position and a lot of other stuff. I think adding the logic stated above would concretely solve the related issues as well. @s77rt also suggested the same (here)[https://github.com/Expensify/App/issues/15810#issuecomment-1465023649]. I think this is worth a shot.

**Comment by tienifr** [comment]
@parasharrajat I think in [the solution](https://github.com/Expensify/App/issues/15194#issuecomment-1465221995) we're not really doing anything shady here, the solution is valid (not a hack) and the only reason we need to put in in CSS is because of the library limitation.

if we think it's a good solution then we shouldn't let a library limitation hold us off.

> Unfortunately, this can't be done unless we make an exception for it.

cc @marcochavezf any thought on this?

**Comment by MelvinBot** [comment]
Current assignee @parasharrajat is eligible for the Internal assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
@marcochavezf, @kevinksullivan, @parasharrajat Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by kevinksullivan** [comment]
@parasharrajat still needs to review some of the other proposals. @parasharrajat once you've done so let us know if anything passes, otherwise we'll need to take this internal.

**Comment by MelvinBot** [comment]
@marcochavezf, @kevinksullivan, @parasharrajat Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by marcochavezf** [comment]
Hi @parasharrajat, any update?

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $4000

**Comment by marcochavezf** [comment]
> @parasharrajat I think in [the solution](https://github.com/Expensify/App/issues/15194#issuecomment-1465221995) we're not really doing anything shady here, the solution is valid (not a hack) and the only reason we need to put in in CSS is because of the library limitation.
> 
> if we think it's a good solution then we shouldn't let a library limitation hold us off.
> 
> > Unfortunately, this can't be done unless we make an exception for it.
> 
> cc @marcochavezf any thought on this?

Could you expand on the library limitation? We aim to fix the root cause instead of band-aid solutions, so if it's a library limitation, should we fix it in the library?

**Comment by marcochavezf** [comment]
> Using the "float" property might be concerning, but we can achieve the same result as "float: left" by applying the `display: flex` can add `styles.dFlex` property to it's parent `Text`. which separates the editable text from the parent text and fixes the selection. I tested it and found that it completely fixes the issue.

Hi @getusha, just to be sure, this is a different solution than the one proposed [here](https://github.com/Expensify/App/issues/15194#issuecomment-1460710278), correct?

**Comment by tienifr** [comment]
> @tienifr could you expand on the library limitation? We aim to fix the root cause instead of band-aid solutions, so if it's a library limitation, should we fix it in the library?

@marcochavezf the limitation is that React does not support targeting pseudo-elements using inline-style (see [here](https://stackoverflow.com/a/45730546) for detail). This is because React inline style uses the `style` html attribute under the hood, which does not support targeting pseudo-elements inline.
I don't think we can fix anything in the library since it's like that by definition.

This, however, doesn't mean using pseudo-elements is a workaround. Pseudo-elements are valid HTML feature, just that it cannot be targeted by using inline style. For other pseudo-elements what we're currently doing is we set it in CSS, for example [here](https://github.com/Expensify/App/pull/16005/files) where we use the `:placeholder` pseudo-element.

FYI Github is also using the exact [same solution I mentioned](https://github.com/Expensify/App/issues/15194#issuecomment-1465221995) for that use case.

**Comment by s77rt** [comment]
The root cause is browser related ([Selection API](https://developer.mozilla.org/en-US/docs/Web/API/Selection)). I think we should fix that on the [SelectionScraper](https://github.com/Expensify/App/blob/main/src/libs/SelectionScraper/index.js) level.

Pseudo-elements are meant for decoration purposes only; making a text unselectable should not implicit making it a decoration e.g. on https://github.com/Expensify/App/issues/15810 we have same issue with emojis, it won't make much sense to treat those as a decoration.

## Proposal
[Cross-Proposal](https://github.com/Expensify/App/issues/15810#issuecomment-1465023649)

**Comment by tienifr** [comment]
I actually think both that edited and emojis (that should be unselectable and uncopiable) are in fact “decorations”, since they are not “page content” (which should be selectable and copiable) 👍

**Comment by MelvinBot** [comment]
@marcochavezf, @kevinksullivan, @parasharrajat Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@marcochavezf, @kevinksullivan, @parasharrajat Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by marcochavezf** [comment]
Hi @parasharrajat, any updates on this?

**Comment by tienifr** [comment]
Just a small note, any solution accepted here will be applied to [this issue](https://github.com/Expensify/App/issues/15810) as well . If we go with manipulating the style to avoid copying we likely need to do it again on a case by case basis for every copy-related issues.

**Comment by mdneyazahmad** [comment]
@parasharrajat what's wrong with this proposal https://github.com/Expensify/App/issues/15194#issuecomment-1444045148. Maybe, I am missing something, but I do not see a review comments about that proposal. I think that solves the issue without any workaround. In fact, I have the exact same proposal, but I can not post a duplicate one.

**Comment by jayeshmangwani** [comment]
## Proposal

[Updated](https://github.com/Expensify/App/issues/15194#issuecomment-1444045148)

hey @parasharrajat I have updated the depth details in my proposal

Thanks @mdneyazahmad , for reminding my proposal, I wasn't active in this issue's conversation and I forgot about this issue.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @arielgreen (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by kevinksullivan** [comment]
@arielgreen heading OOO so going to pass this one over to you. Thanks!

**Comment by MelvinBot** [comment]
@marcochavezf, @parasharrajat, @arielgreen Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by mdneyazahmad** [comment]
Applying either `flex` or `inline-flex` has this issue when the message is overflowed to multiple lines.

<img width="1058" alt="Screenshot 2023-04-19 at 11 18 22 AM" src="https://user-images.githubusercontent.com/77761491/232978565-23706e43-11ad-4dec-89df-4f7ce93b9920.png">

**Comment by s77rt** [comment]
@Ollyws @marcochavezf Won't the proposal break triple-click selection as mentioned here https://github.com/Expensify/App/issues/15194#issuecomment-1460733406? Triple click on the empty space next to `(edited)` and verify the text is selected.

**Comment by s77rt** [comment]
@Ollyws I'm not sure we want to change the hierarchy even if we can, we will struggle with similar issues. Example: The bug would still be reproducible with reaction emojis and those cannot be moved.

Check video below, just verify that the results you get are as shown in the second part of the video

https://user-images.githubusercontent.com/16493223/224516571-a2ba685d-55af-4317-8f75-5394580ce38c.mp4

**Comment by s77rt** [comment]
@getusha Not on the edited label but on the empty space.

@Ollyws I meant to refer to the reactions emoji:
1. Send a message
2. React to that message
3. Triple click selection (on an empty space)
4. Copy the message
5. Verify the copied message does not include the edited label nor emojis from context menu nor the emojis from the reactions buttons.

**Comment by marcochavezf** [comment]
Hi @Ollyws, yes please, let's also consider the [fix](https://github.com/Expensify/App/issues/15194#issuecomment-1514377262) for the emoji issue in the PR. Also, thank you @getusha @s77rt @mdneyazahmad for your comments and for pointing out the edge cases

**Comment by MelvinBot** [comment]
@marcochavezf, @Ollyws, @parasharrajat, @arielgreen Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by marcochavezf** [comment]
PR still in review

**Comment by MelvinBot** [comment]
⚠️ Looks like this issue was linked to a **Deploy Blocker** [here](https://github.com/Expensify/App/issues/18358)

If you are the assigned CME please investigate whether the linked PR caused a regression and leave a comment with the results.

If a regression has occurred and you are the assigned CM follow the instructions [here](https://stackoverflow.com/c/expensify/questions/13647).

If this regression could have been avoided please consider also proposing a recommendation to the PR checklist so that we can avoid it in the future.

**Comment by kaushiktd** [comment]
I doubt using style={{userSelect: 'none'}} and selectable={false}, both will work. 

When copying the selected text, it will copy the whole string with edited text and quick emojis. Please correct me if I am missing something here. @allroundexperts @dhairyasenjaliya 

![Screenshot 2023-05-04 at 9 50 43 AM](https://user-images.githubusercontent.com/43398804/236112420-8e51311a-775a-4fc0-b212-b5a1c79dd4f9.png)

**Comment by marcochavezf** [comment]
> @marcochavezf @parasharrajat The [commit](https://github.com/Expensify/App/pull/17704/commits/7633f88b31b75f24ace3b70f03226558f157da84) that caused that [deploy blocker ](https://github.com/Expensify/App/issues/18358) is actually no longer necessary and can be removed. What's the process here?

@Ollyws yeah, let's remove that commit and create the PR again

**Comment by arielgreen** [comment]
Upwork job auto closed, here is the new one: https://www.upwork.com/jobs/~011daec0b1f8688fbb

**Comment by arielgreen** [comment]
@Ollyws can you please apply to the posting here: https://www.upwork.com/jobs/~011daec0b1f8688fbb

@dhairyasenjaliya @parasharrajat offers sent in Upwork

**Comment by arielgreen** [comment]
@Ollyws offer sent

**Comment by arielgreen** [comment]
Discussing with Marco [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1684461142030209)

**Comment by marcochavezf** [comment]
Last [PR](https://github.com/Expensify/App/pull/18523) has been merged but has not been deployed yet.

**Comment by arielgreen** [comment]
Merged [May 19](https://github.com/Expensify/App/pull/18523#event-9291683692), updating payment date accordingly.

**Comment by arielgreen** [comment]
Reassigning, OOO tomorrow and Monday.

**Comment by twisterdotcom** [comment]
Just waiting on you @parasharrajat for the contract: https://www.upwork.com/nx/wm/offer/24481238

And the BZ Checklist: https://github.com/Expensify/App/issues/15194#issuecomment-1562078430

**Comment by twisterdotcom** [comment]
Payment sent.

**Comment by twisterdotcom** [comment]
Test Case addition requested.
